pydicom/tests/test_uid.py

```diff
@@ -112,25 +40,180 @@ def testGenerateUID(self):
             self.assertEqual(uid[:len(valid_prefix)], valid_prefix)
             self.assertTrue(len(uid) <= 64)
 
-    def testIsValid(self):
+
+class TestUID(unittest.TestCase):
+    """Test DICOM UIDs"""
+    def setUp(self):
+        """Set default UID"""
+        self.uid = UID('1.2.840.10008.1.2')
+
+    def test_equality(self):
+        """Test that UID.__eq__ works with private UIDs."""
+        self.assertTrue(self.uid == UID('1.2.840.10008.1.2'))
+        self.assertTrue(self.uid == '1.2.840.10008.1.2')
+        self.assertTrue(self.uid == 'Implicit VR Little Endian')
+        self.assertFalse(self.uid == UID('1.2.840.10008.1.2.1'))
+        self.assertFalse(self.uid == '1.2.840.10008.1.2.1')
+        self.assertFalse(self.uid == 'Explicit VR Little Endian')
+        # Issue 96
+        self.assertFalse(self.uid == 3)
+        self.assertFalse(self.uid is None)
+
+    def test_inequality(self):
```

Comment:
Yeah, I know. I prefer to be thorough when writing tests (see also: test_is_encapulated vs test_is_compressed).